runtime/compiler/env/j9method.cpp

```diff
@@ -9412,3 +9418,34 @@ TR_J9ByteCodeIlGenerator::packReferenceChainOffsets(TR::Node *node, std::vector<
    return;
    }
 #endif
+
+bool
+TR_ResolvedJ9Method::isFieldQType(int32_t cpIndex)
+   {
+   if (!TR::Compiler->om.areValueTypesEnabled() ||
+      (-1 == cpIndex))
+      return false;
+
+   J9VMThread *vmThread = fej9()->vmThread();
+   J9ROMFieldRef *ref = (J9ROMFieldRef *) (&romCPBase()[cpIndex]);
+   J9ROMNameAndSignature *nameAndSignature = J9ROMFIELDREF_NAMEANDSIGNATURE(ref);
+   J9UTF8 *signature = J9ROMNAMEANDSIGNATURE_SIGNATURE(nameAndSignature);
+   TR::VMAccessCriticalSection isFieldQType(fej9());
```

Comment:
Why does this have to be done under the lock? Since the string is in the constant pool and we don't need to do resolution to look at it?

Reply:
The lock is not necessary since the API just checks the string. It'll be removed.